### PR TITLE
[ISSUE #4502] Optimizing Log Printing Using the LogUtils (#4502)

### LINKED ISSUE #4502: [Enhancement] Optimizing Log Printing Using the LogUtils
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

Currently in the project, most places determine whether to print logs through conditional checks. Now there is a common utility class LogUtils to handle these conditional checks. This makes the code look much simpler.

### current situation
image

### After settlement
image

**Note: Developers need to judge whether each place needs to be changed. For example, some prints are mandatory and do not need conditional checks, so there is no need to change them.**

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct) *

### PR DESCRIPTION
**Pull request by scwlkq**
Fixes #4502

### Modifications
- Optimizing Log Printing Using the   ##LogUtils.
- Preserved some logic based on log level judgment, such as the following:

### Documentation

- Does this pull request introduce a new feature? (yes / no) no
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: eventmesh-common/src/main/java/org/apache/eventmesh/common/file/WatchFileTask.java [hunk 1]
```diff
@@ -38,10 +40,8 @@ public class WatchFileTask extends Thread {
     private final transient WatchService watchService;
 
     private final transient List<FileChangeListener> fileChangeListeners = new ArrayList<>();
-
-    private transient volatile boolean watch = true;
-
     private final transient String directoryPath;
+    private transient volatile boolean watch = true;
```
Comment (Pil0tXia) [inline eventmesh-common/src/main/java/org/apache/eventmesh/common/file/WatchFileTask.java:45]: Is this modification necessary? The new line was removed.

File: eventmesh-common/src/main/java/org/apache/eventmesh/common/protocol/tcp/codec/Codec.java [hunk 2]
```diff
@@ -50,16 +51,81 @@ public class Codec {
     private static final byte[] CONSTANT_MAGIC_FLAG = serializeBytes("EventMesh");
     private static final byte[] VERSION = serializeBytes("0000");
 
+    private static Object deserializeBody(String bodyJsonString, Header header) throws JsonProcessingException {
+        Command command = header.getCmd();
+        switch (command) {
+            case HELLO_REQUEST:
+            case RECOMMEND_REQUEST:
```
Comment (Pil0tXia) [inline eventmesh-common/src/main/java/org/apache/eventmesh/common/protocol/tcp/codec/Codec.java:58]: Private methods should be located at the very bottom of a class.

File: eventmesh-connectors/eventmesh-connector-jdbc/src/main/java/org/apache/eventmesh/connector/jdbc/connection/JdbcConnection.java [hunk 3]
```diff
@@ -76,6 +73,29 @@ public JdbcConnection(JdbcConfig jdbcConfig, InitialOperation initialOperation,
...
+    /**
+     * Creates a ConnectionFactory that uses a pattern-based URL with placeholder values.
+     *
+     * @param urlWithPlaceholder The URL pattern with placeholders.
+     * @param replaces           The replacement values for the placeholders.
+     * @return The ConnectionFactory instance.
+     */
+    @SuppressWarnings("unchecked")
+    public static ConnectionFactory createPatternConnectionFactory(String urlWithPlaceholder, String... replaces) {
+        return config -> {
```
Comment (Pil0tXia) [inline eventmesh-connectors/eventmesh-connector-jdbc/src/main/java/org/apache/eventmesh/connector/jdbc/connection/JdbcConnection.java:85]: This static method is called by another class, so you shouldn't move its position and it is recommended to leave it where it was at the bottom of this class.

There are other similar places, such as private enums in `MysqlCdcEngine` and those changes should be reverted.

File: eventmesh-connectors/eventmesh-connector-jdbc/src/main/java/org/apache/eventmesh/connector/jdbc/connection/JdbcConnection.java [hunk 4]
```diff
@@ -46,13 +47,9 @@ public class JdbcConnection implements AutoCloseable {
...
     private final JdbcConfig jdbcConfig;
-
-    private volatile Connection connection;
-
     private final InitialOperation initialOperation;
-
     private final ConnectionFactory connectionFactory;
-
+    private volatile Connection connection;
     private JdbcDriverMetaData jdbcDriverMetaData;
```
Comment (Pil0tXia) [inline eventmesh-connectors/eventmesh-connector-jdbc/src/main/java/org/apache/eventmesh/connector/jdbc/connection/JdbcConnection.java:53]: There is no need to change field positions and blank lines are missing.

Also, the `@Slf4j` annotation on L37 of this class should be located below the javadoc.

File: eventmesh-connectors/eventmesh-connector-jdbc/src/main/java/org/apache/eventmesh/connector/jdbc/connection/JdbcConnection.java [hunk 5]
```diff
@@ -432,13 +442,11 @@ public <T> T preparedQuery(String sql, ResultSetMapper<T> resultSetMapper, Prepa
      * @throws SQLException if a database access error occurs.
      */
     public <T> T preparedQuery(String sql, PreparedStatementFactory preparedStatementFactory, ResultSetMapper<T> resultSetMapper,
-        PreparedParameter... preparedParameters) throws SQLException {
+                               PreparedParameter... preparedParameters) throws SQLException {
```
Comment (Pil0tXia) [inline eventmesh-connectors/eventmesh-connector-jdbc/src/main/java/org/apache/eventmesh/connector/jdbc/connection/JdbcConnection.java:446]: Redundant indention here. We rarely use this kind of line break. There are other similar places.

File: eventmesh-connectors/eventmesh-connector-jdbc/src/main/java/org/apache/eventmesh/connector/jdbc/source/dialect/cdc/mysql/MysqlCdcEngine.java [hunk 6]
```diff
@@ -96,12 +96,9 @@
...
 public class MysqlCdcEngine extends AbstractCdcEngine<MysqlAntlr4DdlParser, MysqlJdbcContext, MysqlDatabaseDialect> {
 
+    private final EnumMap<EventType, Consumer<Event>> eventHandlers = new EnumMap<>(EventType.class);
     private BinaryLogClient client;
-
     private BlockingQueue<Event> eventQueue = new LinkedBlockingQueue<>(10000);
-
-    private final EnumMap<EventType, Consumer<Event>> eventHandlers = new EnumMap<>(EventType.class);
-
     private Map<Long/* table number */, TableId> tableIdMap = new HashMap<>(64);
```
Comment (Pil0tXia) [inline eventmesh-connectors/eventmesh-connector-jdbc/src/main/java/org/apache/eventmesh/connector/jdbc/source/dialect/cdc/mysql/MysqlCdcEngine.java:102]: Field sequence and formating problem. There are other similar places.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/util/EventMeshUtil.java [hunk 7]
```diff
@@ -295,20 +288,20 @@ public static void printState(final ThreadPoolExecutor scheduledExecutorService)
...
                 return (T) inputStream.readObject();
             }
         }
 
     }
 
     public static Map<String, Object> getCloudEventExtensionMap(final String protocolVersion,
-        final CloudEvent cloudEvent) {
+                                                                final CloudEvent cloudEvent) {
         final EventMeshCloudEventWriter eventMeshCloudEventWriter = new EventMeshCloudEventWriter();
```
Comment (Pil0tXia) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/util/EventMeshUtil.java:305]: This method signature can be merged into one line.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/util/EventMeshUtil.java [hunk 8]
```diff
@@ -295,20 +288,20 @@ public static void printState(final ThreadPoolExecutor scheduledExecutorService)
...
         try (ByteArrayOutputStream byOut = new ByteArrayOutputStream();
-            ObjectOutputStream outputStream = new ObjectOutputStream(byOut)) {
+             ObjectOutputStream outputStream = new ObjectOutputStream(byOut)) {
 
             outputStream.writeObject(object);
 
             try (ByteArrayInputStream byIn = new ByteArrayInputStream(byOut.toByteArray());
-                ObjectInputStream inputStream = new ObjectInputStream(byIn)) {
+                 ObjectInputStream inputStream = new ObjectInputStream(byIn)) {
                 return (T) inputStream.readObject();
```
Comment (Pil0tXia) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/util/EventMeshUtil.java:297]: I'm afraid your CheckStyle plugin was misconfigured. Please have a check.

@yanrongzhen Is it a side effect of Spotless plugin? This contributor once ran `checkStyleMain` gradle task.

**Review by Pil0tXia — CHANGES_REQUESTED** [review]
Overall great job for a freshman. However, there were some ill-formed modifications introduced. Please revert these non-standard formattings.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/util/EventMeshUtil.java (hunk 8 above)
Comment (yanrongzhen) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/util/EventMeshUtil.java:297]: I think it's not a side effect of Spotless, I tried to perform `checkStyleMain`, but failed to reproduce.

**Review by mxsm — COMMENTED** [review]
@scwlkq Thanks for your contribution! I have a small suggestion that PRs should try to do one thing at a time. So this PR should only handle log processing. I saw that there are many format changes and method moves, as well as code deletions, etc. If necessary, these issues can be handled by raising new ISSues instead of handling them in one PR. Try to avoid doing multiple things in one PR.

**Review by mxsm — CHANGES_REQUESTED** [review]
@scwlkq  This branch has conflicts that must be resolved

**Comment by scwlkq** [comment]
It was my negligence to configure checkcode but not apply it. I will correct the points mentioned above. Thank you for your patience and correction.

File: eventmesh-examples/src/main/java/org/apache/eventmesh/grpc/sub/app/service/SubService.java [hunk 9]
```diff
@@ -48,11 +49,15 @@
...
+    private final CountDownLatch countDownLatch = new CountDownLatch(AsyncPublishInstance.MESSAGE_SIZE);
     private EventMeshGrpcConsumer eventMeshGrpcConsumer;
-
     private Properties properties;
-
-    private final SubscriptionItem subscriptionItem = new SubscriptionItem();
+    private final String localPort = properties.getProperty(SERVER_PORT);
+    private final String url = getURL(localPort, "/sub/test");
+    private final String eventMeshIp = properties.getProperty(ExampleConstants.EVENTMESH_IP);
+    private final String eventMeshGrpcPort = properties.getProperty(ExampleConstants.EVENTMESH_GRPC_PORT);
```
Comment (Pil0tXia) [inline eventmesh-examples/src/main/java/org/apache/eventmesh/grpc/sub/app/service/SubService.java:61]: Please note that there are many other similar changes that need to be reverted, including two types of problems, "redundant indention" and "field/method sequence and formatting", as mentioned in my last review.

**Review by Pil0tXia — CHANGES_REQUESTED** [review]

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/LocalUnSubscribeEventProcessor.java [hunk 10]
```diff
@@ -235,12 +229,12 @@ public void handler(final HandlerService.HandlerSpecific handlerSpecific, final
...
-        return new String[]{RequestURI.UNSUBSCRIBE_LOCAL.getRequestURI()};
+        return new String[] {RequestURI.UNSUBSCRIBE_LOCAL.getRequestURI()};
     }
 
     private void registerClient(final HttpEventWrapper requestWrapper,
-        final String consumerGroup,
-        final List<String> topicList, final String url) {
+                                final String consumerGroup,
+                                final List<String> topicList, final String url) {
         Objects.requireNonNull(requestWrapper, "requestWrapper can not be null");
```
Comment (Pil0tXia) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/LocalUnSubscribeEventProcessor.java:238]: Redundant indention here. These params can be merged into one line.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/UnSubscribeProcessor.java [hunk 11]
```diff
@@ -226,7 +222,7 @@ public void processRequest(final ChannelHandlerContext ctx, final AsyncContext<H
                         UnSubscribeResponseBody.class);
                     final long endTime = System.currentTimeMillis();
                     log.error("message|eventMesh2mq|REQ|ASYNC|send2MQCost={}ms"
-                        + "|topic={}|url={}", endTime - startTime,
+                            + "|topic={}|url={}", endTime - startTime,
                         JsonUtils.toJSONString(unSubscribeRequestBody.getTopics()),
                         unSubscribeRequestBody.getUrl(), e);
```
Comment (Pil0tXia) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/UnSubscribeProcessor.java:227]:
```suggestion
log.error("message|eventMesh2mq|REQ|ASYNC|send2MQCost={}ms|topic={}|url={}", endTime - startTime,
                        JsonUtils.toJSONString(unSubscribeRequestBody.getTopics()), unSubscribeRequestBody.getUrl(), e);
```

The same as L201-L204 in this class.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/push/AsyncHTTPPushRequest.java [hunk 12]
```diff
@@ -76,9 +77,8 @@ public class AsyncHTTPPushRequest extends AbstractHTTPPushRequest {
     public static final Logger CMD_LOGGER = LoggerFactory.getLogger(EventMeshConstants.CMD);
 
     public static final Logger LOGGER = LoggerFactory.getLogger("AsyncHTTPPushRequest");
-
-    public String currPushUrl;
     private final Map<String, Set<AbstractHTTPPushRequest>> waitingRequests;
+    public String currPushUrl;
```
Comment (Pil0tXia) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/push/AsyncHTTPPushRequest.java:82]: Field sequence problem here.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/group/ClientGroupWrapper.java [hunk 13]
```diff
@@ -75,52 +76,33 @@
...
-
     private final EventMeshTCPServer eventMeshTCPServer;
-
-    private TcpRetryer tcpRetryer;
-
-    private EventMeshTcpMonitor eventMeshTcpMonitor;
-
-    private DownstreamDispatchStrategy downstreamDispatchStrategy;
-
     private final ReadWriteLock groupLock = new ReentrantReadWriteLock();
```
Comment (Pil0tXia) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/group/ClientGroupWrapper.java:80]: Field sequence problem here.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/group/ClientGroupWrapper.java [hunk 14]
```diff
@@ -249,7 +227,7 @@ public boolean removeSubscription(SubscriptionItem subscriptionItem, Session ses
         String topic = subscriptionItem.getTopic();
         if (session == null
             || !StringUtils.equalsIgnoreCase(group,
-                EventMeshUtil.buildClientGroup(session.getClient().getGroup()))) {
+            EventMeshUtil.buildClientGroup(session.getClient().getGroup()))) {
```
Comment (Pil0tXia) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/group/ClientGroupWrapper.java:230]:
```suggestion
if (session == null || !StringUtils.equalsIgnoreCase(group, EventMeshUtil.buildClientGroup(session.getClient().getGroup()))) {
```

The same as L333-L335, L365-L367, L395-L397, L427-L429.

File: eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/consumer/EventMeshGrpcConsumer.java [hunk 15]
```diff
@@ -64,14 +65,11 @@
...
-    private ManagedChannel channel;
     private final EventMeshGrpcClientConfig clientConfig;
-
     private final Map<String, SubscriptionInfo> subscriptionMap = new ConcurrentHashMap<>();
-
     private final ScheduledThreadPoolExecutor scheduler = new ScheduledThreadPoolExecutor(Runtime.getRuntime().availableProcessors(),
         new EventMeshThreadFactory("GRPCClientScheduler", true));
-
+    private ManagedChannel channel;
     private ConsumerServiceBlockingStub consumerClient;
```
Comment (Pil0tXia) [inline eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/consumer/EventMeshGrpcConsumer.java:73]: Field sequence problem here.

File: eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/consumer/SubStreamHandler.java [hunk 16]
```diff
@@ -47,13 +48,11 @@ public class SubStreamHandler<T> extends Thread implements Serializable {
...
-
-    private transient StreamObserver<CloudEvent> sender;
-
     private final ReceiveMsgHook<T> listener;
+    private transient StreamObserver<CloudEvent> sender;
 
     public SubStreamHandler(final ConsumerServiceStub consumerAsyncClient, final EventMeshGrpcClientConfig clientConfig,
-        final ReceiveMsgHook<T> listener) {
+                            final ReceiveMsgHook<T> listener) {
         this.consumerAsyncClient = consumerAsyncClient;
```
Comment (Pil0tXia) [inline eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/consumer/SubStreamHandler.java:56]: Redundant indention and field sequence problem here. 

**I will refrain from further annotating similar questions below. You can independently review them on this page:**

image

**Review by Pil0tXia — CHANGES_REQUESTED** [review]

**Review by mxsm — COMMENTED** [review]
@scwlkq  Please fix CI

File: eventmesh-webhook/eventmesh-webhook-admin/src/main/java/org/apache/eventmesh/webhook/admin/FileWebHookConfigOperation.java [hunk 17]
```diff
@@ -59,13 +60,27 @@ public FileWebHookConfigOperation(final Properties properties) throws FileNotFou
...
+                bw.write(Objects.requireNonNull(JsonUtils.toJSONString(webHookConfig)));
+            } catch (IOException e) {
+                LogUtils.error(log, "write webhookConfig {} to file error", webHookConfig.getCallbackPath());
+                return false;
+            }
+            return true;
+        }
+    }
+
     @Override
```
Comment (Pil0tXia) [inline eventmesh-webhook/eventmesh-webhook-admin/src/main/java/org/apache/eventmesh/webhook/admin/FileWebHookConfigOperation.java:79]: Method sequence problem here.

**Review by Pil0tXia — CHANGES_REQUESTED** [review]

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/RedirectClientBySubSystemHandler.java [hunk 18]
```diff
@@ -127,22 +125,18 @@ public void handle(final HttpExchange httpExchange) throws IOException {
                 }
             } catch (Exception e) {
                 log.error("clientManage|redirectClientBySubSystem|fail|subSystem={}|destEventMeshIp"
-                    +
-                    "={}|destEventMeshPort={},errMsg={}", subSystem, destEventMeshIp, destEventMeshPort, e);
+                    + "={}|destEventMeshPort={}", subSystem, destEventMeshIp, destEventMeshPort, e);
```
Comment (Pil0tXia) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/RedirectClientBySubSystemHandler.java:129]: If you want to remove the last param of a logger, please unify the same logger in other endpoints such as `RedirectClientByPathHandler`.

File: eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/client/common/RequestContext.java [hunk 19]
```diff
@@ -76,9 +77,7 @@ public void finish(Package msg) {
 
     public static RequestContext context(Object key, Package request, CountDownLatch latch) throws Exception {
         RequestContext c = new RequestContext(key, request, latch);
-        if (log.isInfoEnabled()) {
-            log.info("_RequestContext|create|key=" + key);
-        }
+        LogUtils.info(log, "_RequestContext|create|key=" + key);
```
Comment (mxsm) [inline eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/client/common/RequestContext.java:80]: How about replace with:
```java
 LogUtils.info(log, "_RequestContext|create|key={}", key);
```

File: eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/demo/SyncPubClient.java [hunk 20]
```diff
@@ -38,10 +39,8 @@ public static void main(String[] args) throws Exception {
                 Package rr = pubClient.rr(MessageUtils.rrMesssage("TEST-TOPIC-TCP-SYNC", i), 3000);
                 if (rr.getBody() instanceof EventMeshMessage) {
                     String body = ((EventMeshMessage) rr.getBody()).getBody();
-                    if (log.isInfoEnabled()) {
-                        log.info("rrMessage: " + body + "             "
-                            + "rr-reply-------------------------------------------------" + rr);
-                    }
+                    LogUtils.info(log, "rrMessage: " + body + "             "
+                        + "rr-reply-------------------------------------------------" + rr);
```
Comment (mxsm) [inline eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/demo/SyncPubClient.java:43]: Same as above

File: eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/demo/BroadCastSubClient.java [hunk 21]
```diff
@@ -46,9 +47,7 @@ public void handle(Package msg, ChannelHandlerContext ctx) {
                     if (msg.getHeader().getCommand() == Command.BROADCAST_MESSAGE_TO_CLIENT) {
                         if (msg.getBody() instanceof EventMeshMessage) {
                             String body = ((EventMeshMessage) msg.getBody()).getBody();
-                            if (log.isInfoEnabled()) {
-                                log.info("receive message -------------------------------" + body);
-                            }
+                            LogUtils.info(log, "receive message -------------------------------" + body);
```
Comment (mxsm) [inline eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/demo/BroadCastSubClient.java:50]: Same as above

File: eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/producer/EventMeshGrpcProducer.java [hunk 22]
```diff
@@ -60,9 +61,7 @@ public EventMeshGrpcProducer(EventMeshGrpcClientConfig clientConfig) {
     }
 
     public <T> Response publish(T message) {
-        if (log.isInfoEnabled()) {
-            log.info("Publish message " + message.toString());
-        }
+        LogUtils.info(log, "Publish message " + message.toString());
```
Comment (mxsm) [inline eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/producer/EventMeshGrpcProducer.java:64]: Same as above

**Review by Pil0tXia — APPROVED** [review]
LGTM

**Review by mxsm — DISMISSED** [review]

**Review by mxsm — APPROVED** [review]

**Review by xwm1992 — APPROVED** [review]

**Pull request merged**